classes/steps/actions/role_assign_action_step.php

```diff
@@ -0,0 +1,167 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Role assignment action step class.
+ *
+ * @package    tool_trigger
+ * @copyright  Dmitrii Metelkin <dmitriim@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_trigger\steps\actions;
+
+defined('MOODLE_INTERNAL') || die;
+
+class role_assign_action_step extends base_action_step {
+
+    use \tool_trigger\helper\datafield_manager;
+
+    /**
+     * The fields suplied by this step.
```

Comment:
should be "supplied" (who knows how many steps this is wrong in) :)

Reply:
haha :) will fix them all :)

Reply:
Fixed in 10 places